lms/djangoapps/courseware/views/index.py

```diff
@@ -482,12 +485,20 @@ def _create_courseware_context(self, request):
                 table_of_contents['previous_of_active_section'],
                 table_of_contents['next_of_active_section'],
             )
-            courseware_context['unit'] = section_context.get('activate_block_id', '')
+            unit_id = section_context.get('activate_block_id', '')
+            courseware_context['unit'] = unit_id
             courseware_context['fragment'] = self.section.render(self.view, section_context)
 
             if self.section.position and self.section.has_children:
                 self._add_sequence_title_to_context(courseware_context)
 
+            if show_courseware_mfe_link(request.user, staff_access, self.course.id):
+                courseware_context['microfrontend_link'] = create_microfrontend_link(
+                    self.course.id, self.section.location, unit_id
+                )
+        else:
```

Comment:
the `else` doesn't line up with the `if` block?

Reply:
It's the `else` of the `if self.section:` block, because we do a bunch of things only if there's a section specified. (line 469)

Reply:
I'll separate it into a smaller if/else block to make it clearer.